ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java

```diff
@@ -74,52 +93,57 @@ public SchemaResult getValueSchema(
       }
 
       final ParsedSchema schema = srClient.getSchemaBySubjectAndId(subject, id);
-      return fromParsedSchema(topicName, id, schema, expectedFormat);
+      return fromParsedSchema(topicName, id, schema, expectedFormat, isKey);
```

Comment:
Same question as above -- does it make sense to have that type of validation in the schema supplier, rather than in the engine? Seems like we'd be duplicating validation logic by adding it here as well.

Reply:
I think it makes sense to include it here but not in the schema injector (which is why I said perhaps this should be done in SchemaSupplier). The reason for that is that there are two separate pieces of logic: (1) how do we translate a connect schema to a ksqlDB schema and (2) do we support multi-column ksqlDB key schemas?

To me, they're different because I could decided that (1) should either (a) always convert a schema into a single column struct or (b) always convert a struct schema to multiple columns. That's orthogonal to how we decide (2) - in fact we could support (2) with either 1a or 1b. 

The reason I think we need the assertion here for this PR is that we haven't chosen between 1a and 1b - so it makes sense to fail early.

Reply:
The other reason is that the error here can make more sense. We can fail with "your schema in schema registry has multiple columns, we do not support inference for such schemas..." vs. getting an error at the engine "ksqlDB does not support multiple key columns"

Reply:
Sure, I'm convinced -- updated the schema supplier to fail if multiple key columns are found.